script.js

```diff
@@ -1,4 +1,107 @@
+var totalgames = 0;
+var userwins = 0;
+var tiedgames = 0;
+var computerwins = 0;
+var userName = "";
+var currentGameMode = "Waiting for User Name";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+  if (currentGameMode == "Waiting for User Name") {
+    userName = input; // allow user to input userName
+    currentGameMode = "diceGame"; // toggle game mode to play game
+    myOutputValue = "Hello " + userName + ". Lets Play!";
+    return myOutputValue;
+  } else if (currentGameMode == "diceGame") {
+    if (input != "Scissors" || input != "Paper" || input != "Stone") {
+      // Validating the input
+      var myOutputValue = "Only Scissors, Paper or Stone! <br> Try again";
+      var computerchose = computermove();
+      totalgames = totalgames + 1;
+    }
+    // Tied Grouping
```

Comment:
Since after checking the input , if it is not either "Scissors" , "Paper"  and   "Stone". You want to show  "Only Scissors, Paper or Stone! <br> Try again" message , you should have a return myOutputValue and
  var computerchose = computermove();
      totalgames = totalgames + 1;
      should not be in  there as the totalgame will keep increasing even if the input in wrong. 
      
     // It should be

      if (input != "Scissors" && input != "Paper" && input != "Stone") {
      // Validating the input
      var myOutputValue = "Only Scissors, Paper or Stone! <br> Try again";
     return myOutputValue
    }
    var computerchose = computermove();
     totalgames = totalgames + 1;